### PR TITLE
Fix projections for color images

### PR DESCRIPTION
**Pull request by mcb64**
If the image is color, we still want to work on the grayscale image for projections, otherwise we get into math error problems.

Also, close a file, as Zach pointed out.

### CODE REVIEW

File: pycaqtimage/pycaqtimage.sip [hunk 1]
```diff
@@ -540,11 +540,13 @@ static void _computeRoiProj(ImageBuffer* imageBuffer, QRectF* rectRoi, bool bPro
...
     uint64_t  u64PixelSqSum         = 0;
     uint32_t* pPixelLineStart       = pImgValue + y1 * width + x1;
+    int isColor = imageBuffer->isColor && !imageBuffer->useGray;
 
     for (int iY = y1; iY <= y2; ++iY, pPixelLineStart += width) {
 	uint32_t* pPixel = pPixelLineStart;
 	for (int iX = x1; iX <= x2; ++iX, ++pPixel) {
-	    int iValue = *pPixel;
+#define SUMRGB(x) (((x)&0xff)+(((x)>>8)&0xff)+(((x)>>16)&0xff))
+	    int iValue = isColor ? SUMRGB(*pPixel) : *pPixel;
```
Comment (klauer) [inline pycaqtimage/pycaqtimage.sip]: Not unsigned/`uint32_t`?

File: pycaqtimage/pycaqtimage.sip [hunk 2]
```diff
@@ -540,11 +540,13 @@ static void _computeRoiProj(ImageBuffer* imageBuffer, QRectF* rectRoi, bool bPro
     uint64_t  u64PixelSum           = 0;
     uint64_t  u64PixelSqSum         = 0;
     uint32_t* pPixelLineStart       = pImgValue + y1 * width + x1;
+    int isColor = imageBuffer->isColor && !imageBuffer->useGray;
 
     for (int iY = y1; iY <= y2; ++iY, pPixelLineStart += width) {
 	uint32_t* pPixel = pPixelLineStart;
 	for (int iX = x1; iX <= x2; ++iX, ++pPixel) {
-	    int iValue = *pPixel;
+#define SUMRGB(x) (((x)&0xff)+(((x)>>8)&0xff)+(((x)>>16)&0xff))
```
Comment (klauer) [inline pycaqtimage/pycaqtimage.sip:548]: This says "sum the R G and B bytes of the pixel (the lower 24-bits of this unsigned integer) into a single value"

**Review by klauer — APPROVED** [review]
A small PR to camviewer? Inconceivable

Looks good to me 👍

File: pycaqtimage/pycaqtimage.sip (hunk 2 above)
Comment (mcb64) [reply pycaqtimage/pycaqtimage.sip:548]: Yep.  If you take a look a bit higher... 

[https://github.com/mcb64/camviewer/blob/c054558ad04f4cacf081ce63f43bf6410dc844ef/pycaqtimage/pycaqtimage.sip#L338-L339](url)

So are basically choosing to take something that we earlier did RGB and now do GRAY.

File: pycaqtimage/pycaqtimage.sip (hunk 1 above)
Comment (mcb64) [reply pycaqtimage/pycaqtimage.sip]: Point taken.

**Pull request merged**